### ISSUE TITLE
find_isoswitch - from wf-epi2me single-cell transcript matrix

### ISSUE THREAD

**Issue by reJELIN** [Genentech/Isosceles#5]
Hello,

thank you for the isosceles package. I was wondering if it was possible to work from nanopore wf-epi2me single-cell pipeline results with isosceles ?

special emphasis about "find_isoswitch" function that would be great if I didn't need "the Isosceles transcript" in order to make it work.

Best regards

**Comment by mkabza** [comment]
Hi @reJELIN,
Thanks for your question! It looks like the bam files output by wf-single-cell should be compatible with Isosceles if you run wf-single-cell using the '--merge_bam True' flag and then deduplicate the merged bam file with UMITools.  You can then provide that bam file and specify the 'CB' barcode tag to the `bam_to_tcc` function in Isosceles.  That should enable you to use all the downstream quantifications and capabilities from Isosceles.  We'll get back to you with the details and confirmation of the quantifications once we've tested them.

**Comment by RJ0wly** [comment]
Hello thank you for the answer, I figure it out by myself. I wanted to point it out. 

Also with the current  wf-single cell solution if  we want to use the "find_isoswitch" function we need to tweak it bit because we don't have the data$compatible_tx column if we're using the expression transcript counts matrix from wf-single-cell.

I am going to compare the results between starting from bam and using the expression transcript matrix from wf-single-cell directly.

I will update my results soon !

**Comment by reJELIN** [comment]
Hello thank you for the answer, I figure it out by myself. I wanted to point it out.

Also with the current wf-single cell solution if we want to use the "find_isoswitch" function we need to tweak it bit because we don't have the data$compatible_tx column if we're using the expression transcript counts matrix from wf-single-cell.

Also is it normal that when following the isosceles tutorial that the number of barcodes in comparison from the counts matrix of wf-single-cell is drastically different ? and I still didn't get ride of bad quality barcodes. I get something like 839 barcodes and with wf-single-cell I'm capturing way more barcodes (~ 3000 barcodes for my sample)

**Comment by mkabza** [comment]
Hi @reJELIN ,

While testing the usage of BAM files generated by wf-single-cell with Isosceles, we found that only a small minority of spliced reads (1.7%) matched the reference annotations, compared to 88.3% for Sicelore we've been using too far. Manual inspection of the reads indicated that the issue seems to be related to [a reported bug](https://github.com/epi2me-labs/wf-single-cell/issues/78). Until the bug is fixed, we don't recommend using wf-single-cell results as an input for Isosceles.

**Comment by reJELIN** [comment]
Indeed, you're right. It is good to know. Thank you for highligthing this bug that I wasn't aware of.

What is perhaps surprising, is the transcript matrix provided by the epi2me-lab seems quite complete for my data.  I can only suppose that Flames and Isosceles are too different to be compare.

Nonetheless I tweaked your "find_isoswitch" function in order to use epi2me-labs transcript matrix and it worked quite nice for my data. As you can see:

![image_2024-02-29_094903451](https://github.com/timbitz/Isosceles/assets/112702898/8a30bea1-707b-4f55-87c2-77fde0683522)

**Comment by mkabza** [comment]
Hi @reJELIN ,

The bug in the wf-single-cell workflow has been fixed, and its output can be safely used as input for Iosceles - we've added a section describing how to do it to the main vignette ([link](https://timbitz.github.io/Isosceles/docs/Isosceles.html#3_Preparing_single-cell_data)).
